src/Modules/ModuleTypes.jl

```diff
@@ -110,6 +112,18 @@ option is set in suitable functions.
     r.outgoing = WeakKeyIdDict{ModuleFP, Tuple{SMat, Any}}()
     return r
   end
+  
+  function FreeMod{T}(n::Int, R::AdmissibleModuleFPRing, symbol_fun::Function) where T <: AdmissibleModuleFPRingElem
+    r = new{elem_type(R)}()
+    r.n = n
+    r.R = R
+    r.S = symbol_fun
+    r.d = nothing
```

Comment:
Orthogonal to this PR, but if you did this in each constructor:
```suggestion
    r = new{elem_type(R)}(R, n, symbol_fun, nothing)
```
... then Julia would be able to prove for this type that all instances have e.g. `r.R` always bound and can eliminate checks for this wherever `r.R` is accessed. (This won't add up to much compared to e.g. the problems with the orderings, though)